### PR TITLE
Issues/11 - CICD & Linter/Formatter

### PR DESCRIPTION
**Pull request by verySmolBrain**
# Features

## Scripts

 [poethepoet](https://pypi.org/project/poethepoet/) is used to manage custom scripts.

Available development scripts are:

- `poetry run poe format` - Formats the code
- `poetry run poe format_check` - Checks if code is formatted
- `poetry run poe check` - Perform a suite of static checks
- `poetry run poe fix` - Apply formatting and static checks fixes
- `poetry run poe test` - Runs the tests with coverage report

You can run the following command to automatically apply `format` and `check` before each commit and push:

- `poetry run pre-commit install` - Installs pre-commit hooks

Installing the VSCode Ruff extension enables auto-linting and warnings for code smells/common errors.

## CI/CD

Example [here](https://github.com/verySmolBrain/checkers-bot-tournament/actions/runs/12213681436/job/34073565409). Runs on each PR:

- Static Check
- Tests

### CODE REVIEW

**Comment by trentzz** [comment]
Is there any way to make allowed line length longer? Some of the formatting makes things look a little disgusting (see bruh comment lol)

**Comment by trentzz** [comment]
Also I changed it to merge to `add-your-bot-here` instead

File: checkers_bot_tournament/main.py [hunk 1]
```diff
@@ -11,51 +12,36 @@ def main():
...
+        help="Name or path of the bot to use (required in 'one' mode).",
     )
 
-    parser.add_argument(
-        "bot_list",
-        type=str,
-        nargs="+",
-        help="List of bots"
-    )
+    parser.add_argument("bot_list", type=str, nargs="+", help="List of bots")
```
Comment (donren-leung) [inline checkers_bot_tournament/main.py:24]: naw old version was better imo

**Comment by trentzz** [comment]
I'm gonna merge into a branch for the issue brain made, and I can try and tweak some of the linting settings from there.

**Pull request merged**